### PR TITLE
[IconButton] Add animation on mouseover

### PR DESCRIPTION
**Pull request by ricardomv**

### CODE REVIEW

File: modules/Material/Action.qml [hunk 1]
```diff
@@ -69,6 +69,11 @@ QtObject {
     property bool visible: true
 
     /*!
+       Set to \c true to rotate the icon on mouse hover.
```
Comment (iBeliever) [inline modules/Material/Action.qml]: Please add the number of degrees and the direction in which the rotation will happen to help make the documentation a bit more clear.

File: modules/Material/IconButton.qml [hunk 2]
```diff
@@ -54,12 +55,19 @@ Item {
         onClicked: {
             iconButton.clicked()
         }
+
+        onEntered: if (iconButton.hoverAnimation) icon.rotation = 90
```
Comment (iBeliever) [inline modules/Material/IconButton.qml]: This can be simplified by adding the following directly to the Icon:

    rotation: ink.containsMouse ? 90 : 0

**Comment by iBeliever** [comment]
Nice feature! :+1:

File: modules/Material/IconButton.qml (hunk 2 above)
Comment (ricardomv) [inline modules/Material/IconButton.qml]: yes, thats better

File: modules/Material/Action.qml (hunk 1 above)
Comment (ricardomv) [inline modules/Material/Action.qml]: added the number of degrees i don't think the direction is very important, we might need to add more customization to this for icons that are not simetric we should rotate 180 deg or 360

**Pull request closed without merging**